lib/request.js

```diff
@@ -0,0 +1,34 @@
+module.exports = request
+
+var Promise = require('./utils/promise')
+
+var internals = module.exports.internals = {}
+internals.request = require('./utils/request')
+
+function request (state, options) {
+  if (!options) {
+    return Promise.reject(new Error('hoodie.request: URL or options argument must be defined'))
+  }
+
+  var requestOptions = {
+    url: options.url || options
+  }
+
+  if ((/^\/(?!\/)/).test(requestOptions.url)) {
+    requestOptions.url = state.url + requestOptions.url
+  }
+
+  if (options.data) {
+    requestOptions.body = options.data.toString()
+  }
+
+  if (options.method) {
+    requestOptions.method = options.method
+  }
+
+  if (options.headers) {
+    requestOptions.headers = options.headers
+  }
```

Comment:
I think for `options.method` and `options.headers` we don’t need the `if (...)`s. This will do the same I think?

```
requestOptions.method = options.method
requestOptions.headers = options.headers
```